### PR TITLE
Locallyconnected1D

### PR DESCRIPTION
**Pull request by songhappy**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this patch)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+      }
+
+      (0 to batchSize-1).map( i=> {
+
+        results(i) = Engine.model.invoke(() => {
+
+          val inputSample = input.select(1, i + 1)
+          val outputSample = output.select(1, i + 1)
+          // Add bias first
+          (1 to nOutputFrame).map( j=> {
```
Comment (psyyz10) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala]: while loop may make this faster

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+            biasWindow = bias.select(1, j)
+            outputWindow = outputSample.select(dimSeq - 1, j) // setup up bias for each ouputframe
+            outputWindow.copy(biasWindow)
+          })
+
+          // Add the convolution part
+
+          (0 to nOutputFrame -1).map( j=>{
```
Comment (psyyz10) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala]: while loop may make this faster

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+    gradInput.resizeAs(input)
+    gradInput.zero()
+
+      val batchSize = input.size(1)
+      val pageSize = weight.size(2) * weight.size(3)
+
+      var gradOutputSample = Tensor[T]()
+      var gradInputSample = Tensor[T]()
+
+     (0 to batchSize -1).map( i=> {
```
Comment (psyyz10) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala]: while loop may make this faster

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+
+          results(i) = Engine.model.invoke(() => {
+
+            gradInputSample = gradInput.select(1, i + 1)
+            gradOutputSample = gradOutput.select(1, i + 1)
+            var nOutputSampleFrame = nOutputFrame
+
+            var j = 0
+
+            (0 to nOutputSampleFrame - 1).map(j => {
```
Comment (psyyz10) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LocallyConnected1D.scala]: while loop may make this faster

**Comment by songhappy** [comment]
@psyyz10 thanks for reviewing, modified the code according to your comments.
Can you approve it?
jenkins passed.
http://172.168.2.101:8080/job/BigDL-PR-V-Spark-1.6/427/

**Comment by wzhongyuan** [comment]
@songhappy   pls add serialization unit test

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/LocallyConnected1DSpec.scala [hunk 5]
```diff
@@ -0,0 +1,123 @@
+// scalastyle:off
```
Comment (psyyz10) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/LocallyConnected1DSpec.scala]: Remove this line

**Comment by hkvision** [comment]
Can this be merged?

**Comment by wzhongyuan** [comment]
@hkvision  I didn't see any serialization test , the PR would fail if you run again. please double check

**Comment by songhappy** [comment]
add serialization unit test

**Pull request merged**